server/dashboards/preconfigured/app-insights-web-app.ts

```diff
@@ -0,0 +1,505 @@
+export const config: IDashboardConfig = /*return*/ {
+  id: "app_insights_web",
+  featured: true,
+  name: "Application Insights Web Application",
+  icon: "dashboard",
+  url: "app_insights_web",
+  description: "Dashboard to monitor web apps",
+  preview: "/images/application-insights.png",
+  category: "Web Apps",
+  html: `<div>
+      This is Application Insights dashboard for web apps.</br>
+      This dashboard is built to view all data that is sent to Application Insights like requests,
+      dependencies, exceptions, custom events etc..
+    </div>`,
+  config: {
+    connections: {},
+    layout: {
+      isDraggable: true,
+      isResizable: true,
+      rowHeight: 30,
+      verticalCompact: false,
+      cols: { lg: 12, md: 10, sm: 6, xs: 4, xxs: 2 },
+      breakpoints: { lg: 1200, md: 996, sm: 768, xs: 480, xxs: 0 }
+    }
+  },
+  dataSources: [
+    {
+      id: "timespan",
+      type: "Constant",
+      params: { values: ["24 hours", "1 week", "1 month"], selectedValue: "24 hours" },
+      format: "timespan"
+    },
+    {
+      id: "requests",
+      type: "ApplicationInsights/Query",
+      dependencies: { timespan: "timespan", queryTimespan: "timespan:queryTimespan", granularity: "timespan:granularity" },
+      params: {
+        table: "requests",
+        queries: {
+          duration: {
+            query: ({ granularity }) => {
+              return `
+               where duration > 0 |
+               summarize avg_duration= round(avg(duration) / 1000, 1) by bin(timestamp, ${granularity}) |
+               order by timestamp asc `
+            },
+            format: { type: "timeline", args: { timeField: "timestamp", valueField: "avg_duration" } }
+          },
+          serverrequests: {
+            query: ({ granularity }) => {
+              return `
+                summarize sum = sum(itemCount) by bin(timestamp, ${granularity}), success |
+                order by timestamp asc `
+            },
+            format: {
+              type: "bars",
+              args: { barsField: "timestamp", seriesField: "success", valueField: "sum", threshold: 1 }
+            }
+          }
+        }
+      }
+    },
+    {
+      id: "exceptions",
+      type: "ApplicationInsights/Query",
+      dependencies: { timespan: "timespan", queryTimespan: "timespan:queryTimespan", granularity: "timespan:granularity" },
+      params: {
+        table: "exceptions",
+        queries: {
+          type: {
+            query: ({ granularity }) => {
+              return `summarize count = count() by type `
+            },
+            format: { type: "pie", args: { value: "count", label: "type", maxLength: 20 } }
+          },
+          mapActivity: {
+            query: () => `
+                     extend location=strcat(client_City, ', ', client_CountryOrRegion)
+                    | summarize location_count=count() by location 
```

Comment:
Make sure all lines in a query start in the same row (need one more space in each line)
HAE [Here and everywhere]